skimage/feature/tests/test_canny.py

```diff
@@ -66,3 +69,27 @@ def test_mask_none(self):
         result1 = F.canny(np.zeros((20, 20)), 4, 0, 0, np.ones((20, 20), bool))
         result2 = F.canny(np.zeros((20, 20)), 4, 0, 0)
         self.assertTrue(np.all(result1 == result2))
+
+
+def test_quantile_threshold():
+    image = img_as_float(data.camera()[::50,::50])
+
+    # Correct output produced manually with quantiles
+    # of 0.8 and 0.6 for high and low respectively
+    correct_output = np.array([[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+       [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+       [0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0],
+       [0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0],
+       [0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0],
+       [0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0],
+       [0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+       [0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0],
+       [0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0],
+       [0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0],
+       [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=bool)
+
+    result = F.canny(image, low_threshold=0.6, high_threshold=0.8, quantile_threshold=True)
+
+    assert_equal(result, correct_output)
+
+
```

Comment:
please test that an error is returned if a non-valid quantile (outside of [0, 1]) is given to ``canny``.